Server/src/middleware/authenticateJWT.ts

```diff
@@ -0,0 +1,32 @@
+import * as jwt from 'jsonwebtoken';
+import 'dotenv/config';
+import { Request, Response, NextFunction } from 'express';
+
+//https://stackabuse.com/authentication-and-authorization-with-jwts-in-express-js/
+
+/**
+ * This class acts as the middleware to process an incoming Request to the application and 
+ * verifies token existence
+ */
+export class AuthenticateJWT {
```

Comment:
To my taste JWTAuthenticator is a better name, imo it should be a noun

Reply:
also the file name could start with a capital letter, no? what's the chosen convention for the backend?

Reply:
We didnt have one. This might make it abit cleaner